server/utils/builders/docker-file.ts

```diff
@@ -1,26 +1,25 @@
 import type { WriteStream } from "node:fs";
 import { docker } from "@/server/constants";
+import { prepareBuildArgs } from "@/server/utils/docker/utils";
 import * as tar from "tar-fs";
 import type { ApplicationNested } from ".";
 import { getBuildAppDirectory } from "../filesystem/directory";
-import { createEnvFile } from "./utils";
 
 export const buildCustomDocker = async (
 	application: ApplicationNested,
 	writeStream: WriteStream,
 ) => {
-	const { appName, env } = application;
+	const { appName, buildArgs } = application;
 	const dockerFilePath = getBuildAppDirectory(application);
 	try {
 		const image = `${appName}`;
 		const contextPath =
 			dockerFilePath.substring(0, dockerFilePath.lastIndexOf("/") + 1) || ".";
 		const tarStream = tar.pack(contextPath);
 
-		createEnvFile(dockerFilePath, env);
-
```

Comment:
Why did you remove this function? this is required

Reply:
I left a comment about it in the description.
This is not needed, since envs are already added to the service [here](https://github.com/lorenzomigliorero/dokploy/blob/canary/server/utils/builders/index.ts#L110)

Reply:
@lorenzomigliorero Yes but it's not the same, it's what I explained in the Issue that using an .env file the Dockerfile automatically picks up the environment variables.

in fact you can check how we build the docker image from dokploy and you will see that we use an .env file and it automatically uses it as build and as final image.

the args are used in a Docker statement as `ARG` but when you use the `ENV` statement it will use whatever is in the .env file.

Look this example:
`Dockerfile`
https://github.com/Dokploy/dokploy/blob/canary/Dockerfile

How we use the `.env.production.example` from the repo as a `.env.production` in the CI/CD and that are the env's we are using as a Build and as a Final Image
https://github.com/Dokploy/dokploy/blob/canary/.circleci/config.yml#L32

Reply:
@lorenzomigliorero Yes but there are many projects where the most common thing is that when you use a docker image to have an .env file and many times there are validations that if this file does not exist the build is broken, I include myself because I have run other projects where if this file does not exist it throws an exception, example:

https://github.com/NetSocialOSS/Harbinger/blob/main/main.go#L20

Reply:
Oh yeah, sure, I'll revert the change.
I thought it was related to the _buildargs_ discussion, as you commented [here](https://github.com/Dokploy/dokploy/issues/241#issuecomment-2241377235).

On it!